### PR TITLE
Feature/crud photos

### PR DESCRIPTION
**Pull request by aleksVoit**
added photos routes. fixed healthcheck in docker compose

### CODE REVIEW

File: src/database/db.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 load_dotenv()
 
-DATABASE_URL = os.getenv("DATABASE_URL")
+DATABASE_URL = os.getenv('DATABASE_URL')
```
Comment (ValKurta) [inline src/database/db.py:9]: There is no difference between DATABASE_URL = os.getenv("DATABASE_URL") and DATABASE_URL = os.getenv('DATABASE_URL')

File: src/routes/photos.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        secure=True
+    )
+    r = cloudinary.uploader.upload(file.file,
+                                   public_id=f'PhotoShare/{description[:7].replace(" ", "") +
+                                                           file.filename.replace(".", "")}',
+                                   overwrite=True)
+    url = cloudinary.CloudinaryImage(f'PhotoShare/{description[:7].replace(" ", "") +
+                                                   file.filename.replace(".", "")}') \
+        .build_url(crop='fill', version=r.get('version'))
+
```
Comment (oleksyii) [inline src/routes/photos.py:66]: Please, sanitize with variables
```suggestion
    # Create a clean file identifier
    clean_description = description[:7].replace(" ", "")
    clean_filename = file.filename.replace(".", "")
    public_id = f'PhotoShare/{clean_description}{clean_filename}'
    
    # Upload the file
    upload_response = cloudinary.uploader.upload(
        file.file,
        public_id=public_id,
        overwrite=True
    )
    
    # Generate the Cloudinary image URL
    url = cloudinary.CloudinaryImage(public_id)
```

File: src/routes/photos.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        api_secret=settings.cloudinary_api_secret,
+        secure=True
+    )
+    r = cloudinary.uploader.upload(file.file,
+                                   public_id=f'PhotoShare/{description[:7].replace(" ", "") +
+                                                           file.filename.replace(".", "")}',
+                                   overwrite=True)
+    url = cloudinary.CloudinaryImage(f'PhotoShare/{description[:7].replace(" ", "") +
+                                                   file.filename.replace(".", "")}') \
+        .build_url(crop='fill', version=r.get('version'))
```
Comment (oleksyii) [inline src/routes/photos.py:35]: Sanitize

**Review by oleksyii — CHANGES_REQUESTED** [review]
Sanitize some endpoints and we're good

File: src/repository/photos.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+from src.schemas import PhotoModel, PhotoResponse, TagModel, TagsPhoto
+from datetime import date, timedelta
+
+from fastapi import UploadFile
+
+
+async def create_photo(text: str, url: str, db: Session) -> Photo:
+
+    photo = Photo(
+        user_id=1,
```
Comment (ValKurta) [inline src/repository/photos.py:14]: Need to fix the logic with constant value

File: src/repository/photos.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+    photo = db.query(Photo).filter(Photo.id == photo_id).first()
+    if photo:
+        photo.url = url
+        photo.description = description
+        db.commit()
+    return photo
+
+
+async def delete_photo(photo_id: int, db: Session) -> Photo | None:
```
Comment (ValKurta) [inline src/repository/photos.py:38]: Please return a status message or confirmation that the photo has been deleted.Additionally,please to raise an exception if the photo is not found(this is what we discussed on the call in the evening)

File: src/repository/photos.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+    photo = db.query(Photo).filter(Photo.id == photo_id).first()
+    if photo:
+        db.delete(photo)
+        db.commit()
+    return photo
+
+
+async def change_description(photo_id: int, description: str, db: Session) -> Photo | None:
```
Comment (ValKurta) [inline src/repository/photos.py:47]: Please add return a status message in the end

File: src/repository/photos.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+    photo = db.query(Photo).filter(Photo.id == photo_id).first()
+    if photo:
+        photo.description = description
+        db.commit()
+    return photo
+
+
+async def add_tags(photo_id: int, body: TagsPhoto, db: Session) -> Photo | None:
```
Comment (ValKurta) [inline src/repository/photos.py:56]: Please check that yuo can create many to many here(I can see first())

File: src/repository/photos.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+    photo = db.query(Photo).filter(Photo.id == photo_id).first()
+    if photo:
+        photo.tags = body.tags
+        db.commit()
+    return photo
+
+
+async def search_photos(tag: str, db: Session) -> list[Photo]:
+    photos = db.query(Photo).filter(tag in Photo.tags).all()
```
Comment (ValKurta) [inline src/repository/photos.py:66]: filter(tag in Photo.tags) does not work properly - alchemy does not work directly. we need something likes photos =db.query(Photo).join(Photo.tags).filter(Tag.name == tag).all()

File: src/routes/photos.py [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+from src.conf.config import settings
+
+router = APIRouter(prefix="/photos", tags=["photos"])
+
+
+@router.post('/', response_model=PhotoResponse)
+async def create_photo(file: UploadFile = File(),
+                       description: str = Form(),
+                       db: Session = Depends(get_db)):
+    cloudinary.config(
```
Comment (ValKurta) [inline src/routes/photos.py:23]: Please initialize Cloudinary configuration globally one time before endpoints isntead of declare it in every endpoint

File: src/routes/photos.py [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+        cloud_name=settings.cloudinary_name,
+        api_key=settings.cloudinary_api_key,
+        api_secret=settings.cloudinary_api_secret,
+        secure=True
+    )
+    r = cloudinary.uploader.upload(file.file,
```
Comment (ValKurta) [inline src/routes/photos.py:29]: what if we add try with except

File: src/routes/photos.py [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+                       file: UploadFile = File(),
+                       description: str = Form(),
+                       db: Session = Depends(get_db)):
+    cloudinary.config(
+        cloud_name=settings.cloudinary_name,
+        api_key=settings.cloudinary_api_key,
+        api_secret=settings.cloudinary_api_secret,
+        secure=True
+    )
+    r = cloudinary.uploader.upload(file.file,
```
Comment (ValKurta) [inline src/routes/photos.py:59]: what if we add try with except

File: src/routes/posts.py
Comment (ValKurta) [inline src/routes/posts.py:1]: no changes= do not commit. also, I cannot find it in your branch

File: src/schemas.py [hunk 12]
```diff
@@ -45,3 +46,27 @@ class TokenModel(BaseModel):
...
+
+    class Config:
+        from_attributes = True
+
+
+class TagsPhoto(BaseModel):
+    tags: List[str]
+
+
+class PhotoModel(BaseModel):
```
Comment (ValKurta) [inline src/schemas.py:66]: +max_length
+Optional[str]

File: src/schemas.py [hunk 13]
```diff
@@ -45,3 +46,27 @@ class TokenModel(BaseModel):
...
+    description: str
+
+
+class PhotoResponse(PhotoModel):
+    id: int
+    url: str
```
Comment (ValKurta) [inline src/schemas.py:72]: class Config:
        from_attributes = True

File: src/repository/photos.py (hunk 7 above)
Comment (aleksVoit) [reply src/repository/photos.py:56]: add_tags in process

File: src/repository/photos.py (hunk 6 above)
Comment (aleksVoit) [reply src/repository/photos.py:47]: I think it better to return the photo with a new description, and user will see the photo with new description

File: src/repository/photos.py (hunk 5 above)
Comment (aleksVoit) [reply src/repository/photos.py:38]: the exception 'photo not found' is raised in routes/photos:delete_photo

File: src/repository/photos.py (hunk 4 above)
Comment (aleksVoit) [reply src/repository/photos.py:14]: this I can make when we have user role. for the moment authorization is added, but confirmation email is empty and user cannot authorise to load the photos

File: src/database/db.py (hunk 1 above)
Comment (aleksVoit) [reply src/database/db.py:9]: ok

File: src/routes/photos.py (hunk 2 above)
Comment (aleksVoit) [reply src/routes/photos.py:66]: ok

File: src/routes/photos.py (hunk 9 above)
Comment (aleksVoit) [reply src/routes/photos.py:23]: ok. done

File: src/routes/photos.py (hunk 10 above)
Comment (aleksVoit) [reply src/routes/photos.py:29]: ok. done

File: src/routes/photos.py (hunk 11 above)
Comment (aleksVoit) [reply src/routes/photos.py:59]: ok. done

File: src/routes/photos.py (hunk 2 above)
Comment (aleksVoit) [reply src/routes/photos.py:66]: thank you, gut idea! done

File: src/schemas.py (hunk 12 above)
Comment (aleksVoit) [reply src/schemas.py:66]: ok, good idea, done

File: src/schemas.py (hunk 13 above)
Comment (aleksVoit) [reply src/schemas.py:72]: ok, good idea, done

**Pull request closed without merging**